scijava/scijava-ops/src/main/java/org/scijava/ops/provenance/OpExecutionSummary.java

```diff
@@ -0,0 +1,113 @@
+
+package org.scijava.ops.provenance;
+
+import java.util.UUID;
+
+import org.scijava.ops.OpInfo;
+import org.scijava.ops.impl.DefaultOpEnvironment;
+
+/**
+ * Describes the execution of an Op
+ *
+ * @author Gabe Selzer
+ */
+public class OpExecutionSummary {

```

Comment:
Changes made in 5e04129f. My one concern with that commit is that `DefaultOpHistory.addExecution` (as of this writing) is unable to cache different `Object`s with the same hashcode. This change actually required an alteration to `ProvenanceTest`, as we _were_ able to differentiate two `Double`s of the _same value_ returned by two different `Function`s. Specifically, we had `Double d1` returned by `Function f1`, and `Double d2` returned by `Function f2` (where `d1.doubleValue() == d2.doubleValue()` but `d1 != d2`), where `OpHistory.executionsUpon(d1)` would point to _only_ `f1` and `OpHistory.executionsUpon(d2)` would point to _only_ `f2`. Now, we have `OpHistory.executionsUpon(d1)` pointing _both_ `d1` and `d2` (since `d1.hashCode() == d2.hashCode()`, which is wrong. Don't have a good idea of fixing this though, without adding complexity to the implementation 🙁